spec/models/widget/card_spec.rb

```diff
@@ -45,4 +45,36 @@
       expect(Widget::Card.body).not_to include(page_card)
     end
   end
+
+  context "when is not header card" do
+    it "link_url field is required" do
+      invalid_card = build(:widget_card, header: false, link_url: nil)
+
+      expect(invalid_card).not_to be_valid
+    end
+  end
+
+  context "when is header card" do
+    it "is not required if link_text is not provided" do
+      valid_header = build(:widget_card, :header, link_text: nil)
```

Comment:
```suggestion
      header = build(:widget_card, :header, link_text: nil, link_url: nil)
```

Otherwise the factory will generate a card with a `link_url`.